waddrmgr/address.go

```diff
@@ -88,22 +88,31 @@ type ManagedScriptAddress interface {
 	Script() ([]byte, error)
 }
 
+// publicAddress represents a address. It has only the most basic data common to
+// both managed and un-managed addresses.
+type publicAddress struct {
+	manager    *Manager
+	account    uint32
+	address    *btcutil.AddressPubKeyHash
+	imported   bool
+	internal   bool
+	compressed bool
+	used       bool
+}
+
 // managedAddress represents a public key address.  It also may or may not have
 // the private key associated with the public key.
 type managedAddress struct {
-	manager          *Manager
-	account          uint32
-	address          *btcutil.AddressPubKeyHash
-	imported         bool
-	internal         bool
-	compressed       bool
-	used             bool
+	publicAddress
```

Comment:
I thought it might be logical to separate the structs for loose vs keychain addrs for future e.g if we wanted to add a method specific to loose addrs.